### ISSUE TITLE
DELETE _snapshot/$REPO/* should work even if repo is broken

### ISSUE THREAD

**Issue by DaveCTurner** [elastic/elasticsearch#75612]
Today you can execute `DELETE _snapshot/$REPO/*` to delete all snapshots in a repository, but this works in the same way as deleting all the individual snapshots and in particular it requires the repository to be in a valid and healthy state. In fact there's no need to read anything from the repository if we're deleting everything, we could just write a new empty index file and clean up any dangling blobs.

**Comment by original-brownbear** [comment]
This should for the most part already work as described. If you delete all snapshots then all that must be readable is the latest `index-N` at the repo root. Everything else will pretty much be cleaned up without deeper inspection I believe. If that's not the case it's a bug.

**Comment by DaveCTurner** [comment]
Indeed, but it's quite a twisty path with a number of obstacles, e.g. today you can't delete the source of a clone even if you're also aborting the clone by deleting its target. Arguably that's also a bug, we could fix it for sure, but there might be more like it in future and there seems to be a need for a truly robust fallback to all things of that nature.

And indeed the root index file may indeed not be readable, perhaps through no fault of our own.

**Comment by idegtiarenko** [comment]
Currently `org.elasticsearch.snapshots.SnapshotsService#deleteSnapshots` is relying on `repository.executeConsistentStateUpdate` that internally reads `index*` files. It might be easier to implement separate `truncate` operation that would not depend on presence and correctness of any blob repository files (it might also reuse native object storage primitives to truncate in case no prefix and sufficient privileges).

>if we're deleting everything, we could just write a new empty index file and clean up any dangling blobs.

We should also be careful not to delete unknown unrelated files in case somebody accidentally mount other object storage with unrelated files that is not supposed to be used with elasticsearch (for example a bucket with some other system backups instead of elasticsearch repo).

**Comment by idegtiarenko** [comment]
We should:
* confirm that docs disallow keeping unrelated files inside of snapshot repository
* perform delete of every file with a known extension
* keep in mind already running operations if any
